src/uarch-step.cpp

```diff
@@ -1089,5 +1096,55 @@ static inline void uarchExecuteInsn(UarchState &a, uint32 insn, uint64 pc) {
     throw std::runtime_error("illegal instruction");
 }
 
+static inline bool getIflagsH(uint64 iflags) {
+    return (iflags & IFLAGS_H_MASK) != 0;
+}
+
+static inline bool getIflagsY(uint64 iflags) {
+    return (iflags & IFLAGS_Y_MASK) != 0;
+}
+
+template <typename UarchState>
+uarch_step_status uarch_step(UarchState &a) {
+    // This must be the first read in order to match the first log access in machine::verify_state_transition
+    auto cycle = readCycle(a);
```

Comment:
If I am not wrong, we can't use `auto` here, otherwise the solidity Lua generator will break.We should use explicit types.